spec/app/domain/authentication/authn-oidc/v2/commands/ListProviders_spec.rb

```diff
@@ -0,0 +1,113 @@
+require 'spec_helper'
+
+RSpec.describe('Authentication::AuthnOidc::V2::Commands::ListProviders') do
+
+  let(:provider_context) do
+    ::Authentication::AuthnOidc::V2::Views::ProviderContext.new(
+      client: client
+    )
+  end
+
+  let(:client) do
+    class_double(::Authentication::AuthnOidc::V2::Client).tap do |double|
+      allow(double).to receive(:new).and_return(
+        current_client)
+    end
+  end
+
+  let(:current_client) do
+    instance_double(::Authentication::AuthnOidc::V2::Client).tap do |double|
+      allow(double).to receive(:discovery_information).and_return(endpoint)
+    end
+  end
+
+  let(:endpoint) { double(authorization_endpoint: '"http://test"') }
+
+  let(:res) do
+    [{ name: "foo",
+       redirect_uri: "\"http://test\"?client_id=ConjurClient&response_type=code&scope=openid%20email%20profile" \
+         "&state=foostate&nonce=secret&redirect_uri=http%3A%2F%2Fconjur%2Fauthn-oidc%2Fcucumber%2Fauthenticate",
```

Comment:
Should this test differentiate somehow for foononce and barnonce instead of just "secret" for both? Are there other tests that make sure the nonce is unique each time?

Reply:
Also, if we're generating state in the UI now, it's probably not "foostate" and "barstate". Seems these tests might need some reworking after the events of the past week.